<font size="5">**ABHISHEK KUMAR SINGH**</font>

<font size="5">**2K19/CO/021**</font>

**<font size="8"><center>EXPERIMENT - 6</center></font>**

**AIM:** To implement Friedman Test in python.

**THEORY**

**P-value** 

The P value, or calculated probability, is the probability of finding the observed, or more extreme, results when the null hypothesis (H 0) of a study question is true — the definition of ‘extreme’ depends on how the hypothesis is being tested.
If the P value is less than the chosen significance level then we reject the null hypothesis i.e. accept that our sample gives reasonable evidence to support the alternative hypothesis. It does NOT imply a “meaningful” or “important” difference; that is for us to decide when considering the real-world relevance of our result.

**Friedman Test** 

It is a non-parametric test alternative to the one way ANOVA with repeated measures. It tries to determine if subjects changed significantly across occasions/conditions. For example:- Problem-solving ability of a set of people is the same or different in Morning, Afternoon, Evening. It is used to test for differences between groups when the dependent variable is ordinal. This test is particularly useful when the sample size is very small.


**Assumptions**

1. One group that is measured on three or more different occasions.

2. The group is a random sample from the population.

3. The dependent variable should be measured at the ordinal or continuous level.

4. Samples do NOT need to be normally distributed.



**Null Hypothesis:** There is no significant difference between the given conditions of measurement OR the probability distributions for all the conditions are the same. (Medians are same)

**Alternate Hypothesis:** At least 2 of them differ from each other.

H0 : M1 = M2 = M3 = ..... Mk  ; M= Median

H1 : At least two of them show significant difference.

**Test Statistic**

![Friedman](Friedman.png)

n = total number of subjects/participants.

k = total number of blocks to be measured.

Ri = sum of ranks of all subjects for a block i

**Post Hoc Analysis:** We can find out whether there is a difference in any given pair of experimental conditions if the null hypothesis is rejected using the Post Hoc analysis which can be done using Wilcoxon signed-rank test, Conover’s test etc.. In Wilcoxon test, you can get the results for all pairs also but you will have to make a Bonferroni correction which will change the level of significance to Given level of significance/total number of pairs.

**CODE AND OUTPUT:**

**Importing Libraries**

In [3]:
import pandas as pd
import numpy as np
import scipy.stats as stats

**Loading Dataset**

In [4]:
df = pd.read_csv('Friedman.csv')

In [5]:
df.head()

None  Classical  Dance
0     8          8      7
1     7          6      6
2     6          8      6
3     8          9      7
4     5          8      5

**Friedman Test** 

In [7]:
stats.friedmanchisquare(df['None'], df['Classical'], df['Dance'])

FriedmanchisquareResult(statistic=7.599999999999986, pvalue=0.02237077185616575)

There was a statistically significant difference in perceived effort depending on which type of music was listened to whilst running, χ2(2) = 7.600, p = 0.022.

**LEARNING OUTCOMES**

We learnt about Friedman Test.